controller/forwarding/hyperlane_test.go

```diff
@@ -0,0 +1,475 @@
+// AN "AS IS" BASIS. LICENSOR HEREBY DISCLAIMS ALL WARRANTIES AND CONDITIONS,
+// EXPRESS OR IMPLIED, INCLUDING (WITHOUT LIMITATION) WARRANTIES OF
+// MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE, NON-INFRINGEMENT, AND
+// TITLE.
+
+package forwarding_test
+
+import (
+	"context"
+	"testing"
+
+	hyperlaneutil "github.com/bcp-innovations/hyperlane-cosmos/util"
+	warptypes "github.com/bcp-innovations/hyperlane-cosmos/x/warp/types"
+	"github.com/stretchr/testify/require"
+
+	"cosmossdk.io/log"
+	"cosmossdk.io/math"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/noble-assets/orbiter/controller/forwarding"
+	"github.com/noble-assets/orbiter/testutil"
+	"github.com/noble-assets/orbiter/testutil/mocks"
+	"github.com/noble-assets/orbiter/types"
+	forwardingtypes "github.com/noble-assets/orbiter/types/controller/forwarding"
+	"github.com/noble-assets/orbiter/types/core"
+)
+
+func TestNewHyperlaneController(t *testing.T) {
+	testCases := []struct {
+		name     string
+		logger   log.Logger
+		handler  forwardingtypes.HyperlaneHandler
+		expError string
+	}{
+		{
+			name:    "success - valid controller creation",
+			logger:  log.NewNopLogger(),
+			handler: &mocks.HyperlaneHandler{},
+		},
+		{
+			name:     "error - nil logger",
+			expError: "logger cannot be nil",
+		},
+		{
+			name:     "error - when no Hyperlane handler is provided",
+			logger:   log.NewNopLogger(),
+			expError: "handler",
+		},
+	}
+
+	for _, tC := range testCases {
+		t.Run(tC.name, func(t *testing.T) {
+			controller, err := forwarding.NewHyperlaneController(
+				tC.logger,
+				tC.handler,
+			)
+
+			if tC.expError != "" {
+				require.ErrorContains(t, err, tC.expError)
+				require.Nil(t, controller)
+			} else {
+				require.NoError(t, err)
+				require.NotNil(t, controller)
+			}
+		})
+	}
+}
+
+func TestExtractAttributes_Hyperlane(t *testing.T) {
+	usdnID := make([]byte, 32)
+	copy(usdnID, "usdn id")
+
+	testCases := []struct {
+		name       string
+		forwarding func() *core.Forwarding
+		expHypAttr forwardingtypes.HypAttributes
+		expError   string
+	}{
+		{
+			name:       "error - nil forwarding",
+			forwarding: func() *core.Forwarding { return nil },
+			expError:   "forwarding is not set",
+		},
+		{
+			name: "error - CCTP forwarding",
+			forwarding: func() *core.Forwarding {
+				f, err := forwardingtypes.NewCCTPForwarding(
+					0,
+					testutil.RandomBytes(32),
+					testutil.RandomBytes(32),
+					[]byte{},
+				)
+
+				require.NoError(t, err)
+
+				return f
+			},
+			expError: "invalid type",
+		},
+		{
+			name: "success - valid forwarding",
+			forwarding: func() *core.Forwarding {
+				f, err := forwardingtypes.NewHyperlaneForwarding(
+					usdnID,
+					0,
+					make([]byte, 32),
+					make([]byte, 32),
+					"",
+					math.NewInt(1),
+					sdk.NewInt64Coin("usdn", 1),
+					[]byte{},
+				)
+				require.NoError(t, err)
+
+				return f
+			},
+			expHypAttr: forwardingtypes.HypAttributes{
+				TokenId:           []byte("token id"),
```

Comment:
why is this using `[]byte("token id")` when the hyperlane forwarding is created with `"usdn id"`? How come it's not erroring?